engine/src/main/java/org/camunda/bpm/engine/ExternalTaskService.java

```diff
@@ -398,6 +399,19 @@
    */
   public ExternalTaskQuery createExternalTaskQuery();
 
+  /**
+   * Returns a list of distinct topic names.
+   * Returns an empty list if no topics are found.
+   */
+   List<String> getTopicNames();
+
+  /**
+   * Returns a list of distinct topics based on parameters given.
+   * Returns an empty list if no topics are found.
+   */
+
+  List<String> getTopicNames(GetTopicNamesQueryMapper query);
```

Comment:
Instead of a parameter object, we should have three individual boolean parameters. This is more consistent with other APIs.

Please document the parameters as part of the Javadoc.

Reply:
I'm happy to implement this as:
 getTopicNames(boolean locked, boolean unlocked, boolean withRetriesLeft)

However, I felt this method signature was not very concise or clear as to what it was doing. For example, reading getTopicNames(true,false,false) does not say much about what you are doing.

 In addition, the current sql for querying external tasks will not return anything if you have both "locked" and "unlocked" set to true. This could be changed by creating a new sql query for this method, but then the usage differs from externaltaskquery and may be confusing to users.

Reply:
> However, I felt this method signature was not very concise or clear as to what it was doing. For example, reading getTopicNames(true,false,false) does not say much about what you are doing.

Yes, that's correct. Having individual parameters is consistent with other APIs though, which I value higher in this case.

> In addition, the current sql for querying external tasks will not return anything if you have both "locked" and "unlocked" set to true. This could be changed by creating a new sql query for this method, but then the usage differs from externaltaskquery and may be confusing to users.

This is the same with `GetTopicNamesQueryMapper` though, right? This behavior is fine, we should just make sure to document and test it.